build/nlu-regression.js

```diff
@@ -98,58 +98,134 @@ const runAllTests = async axiosConfig => {
     }
 
     passedTests += testResult.success ? 1 : 0
-    console.log(`(${i++} /${tests.length}) #${test.id}`, 'success: ', testResult.success)
+    console.log(`[${botInfo.id}] (${++i} /${tests.length}) #${test.id}`, 'success: ', testResult.success)
   }
 
   return _.round((passedTests / tests.length) * 100, 1)
 }
 
-const compareScore = async score => {
-  const latestResultsFile = `${nluTestingDir}/src/bot-templates/bp-nlu-regression-testing/latest-results.csv`
+const getPreviousScore = async botInfo => {
+  const latestResultsFile = `${nluTestingDir}/src/bot-templates/${botInfo.template.id}/latest-results.csv`
   const latestResultsContent = fs.readFileSync(latestResultsFile, { encoding: 'utf8' })
   const previousScoreOccurence = latestResultsContent.match(/summary: ((100|\d{1,2})[.]\d{1})?/gm)
   if (!previousScoreOccurence || !previousScoreOccurence[0]) {
-    return false
+    return
   }
 
   const previousScoreString = previousScoreOccurence[0].split(':')[1]
   const previousScore = parseFloat(previousScoreString)
-  console.log(chalk.yellow('Previous Score Was:'), previousScore)
 
-  return score >= previousScore
+  return previousScore
 }
 
-const main = async () => {
-  try {
-    let token = await login()
-    if (!token) {
-      token = await signup()
-    }
-    if (!token) {
-      console.error(chalk.red(chalk.bold('Unable To Login Or Sign Up...')))
+const makeAxiosConfig = token => {
+  return {
+    headers: {
+      Authorization: `Bearer ${token}`,
+      'X-BP-Workspace': 'default'
     }
+  }
+}
 
-    const axiosConfig = {
-      headers: {
-        Authorization: `Bearer ${token}`,
-        'X-BP-Workspace': 'default'
+const getOs = () => {
+  return new Promise((resolve, error) => {
+    getos((err, os) => {
+      if (err) {
+        error(err)
+      } else {
+        resolve(os)
       }
+    })
+  }).catch(_err => ({
+    os: os.platform(),
+    dist: 'default',
+    codename: 'N/A',
+    release: 'N/A'
```

Comment:
not really, but it doesnt hurt... I copied from `src/bp/common/getos.ts`. 

This is only to print a formatted message with the os in it in case there's both a regression and an error when calling getos... (Unlikely to happend).